lib/core/src/system/web/dom/html/css3d_renderer.rs

```diff
@@ -0,0 +1,269 @@
+        let height = format!("{}px", dimensions.y);
+        let doms   = vec![&self.front_dom,&self.back_dom,&self.front_camera,&self.back_camera];
+        for dom in doms {
+            dom.set_property_or_panic("width" , &width);
+            dom.set_property_or_panic("height", &height);
+        }
+    }
+}
+
+// ====================
+// === Css3dRenderer ===
+// ====================
+
+/// A renderer for `Css3dObject`s.
+#[derive(Clone,Debug)]
+pub struct Css3dRenderer {
+    container      : DomContainer,
+    data           : Rc<Css3dRendererData>,
+    logger         : Logger
+}
+
+impl Css3dRenderer {
+    /// Creates a Css3dRenderer inside an element.
+    pub fn from_element<L:Into<Logger>>(logger:L, element:HtmlElement) -> Result<Self> {
+        let logger                     = logger.into();
+        let container                  = DomContainer::from_element(element);
+        let front_dom    : HtmlElement = dyn_into(create_element("div")?)?;
+        let back_dom     : HtmlElement = dyn_into(create_element("div")?)?;
+        let front_camera : HtmlElement = dyn_into(create_element("div")?)?;
+        let back_camera  : HtmlElement = dyn_into(create_element("div")?)?;
+
+        front_dom.set_property_or_panic("position", "absolute");
+        front_dom.set_property_or_panic("top", "0px");
+        front_dom.set_property_or_panic("overflow", "hidden");
+        front_dom.set_property_or_panic("overflow", "hidden");
+        front_dom.set_property_or_panic("width", "100%");
+        front_dom.set_property_or_panic("height", "100%");
+        front_dom.set_property_or_panic("pointer-events", "none");
+        back_dom.set_property_or_panic("position", "absolute");
+        back_dom.set_property_or_panic("top", "0px");
+        back_dom.set_property_or_panic("overflow", "hidden");
+        back_dom.set_property_or_panic("overflow", "hidden");
+        back_dom.set_property_or_panic("width", "100%");
+        back_dom.set_property_or_panic("height", "100%");
+        back_dom.set_property_or_panic("pointer-events", "none");
+        back_dom.set_property_or_panic("z-index", "-1");
+        front_camera.set_property_or_panic("width", "100%");
+        front_camera.set_property_or_panic("height", "100%");
+        front_camera.set_property_or_panic("transform-style", "preserve-3d");
+        back_camera.set_property_or_panic("width", "100%");
+        back_camera.set_property_or_panic("height", "100%");
+        back_camera.set_property_or_panic("transform-style", "preserve-3d");
+
+        container.dom.append_or_panic(&front_dom);
+        container.dom.append_or_panic(&back_dom);
+        front_dom.append_or_panic(&front_camera);
+        back_dom.append_or_panic(&back_camera);
+
+        let data             = Css3dRendererData::new(front_dom, back_dom, front_camera, back_camera);
+        let data             = Rc::new(data);
+        let mut renderer     = Self {container,data,logger};
+
+        renderer.init_listeners();
```

Comment:
You would get a much nicer API if `init_listeners` would ocnume self. This way you could even write 
`let renderer = Self{container,data,logger}.init();` :)